pkg/bridgeclient/websocket_message_handler.go

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) The One True Way 2022. Apache License 2.0. The authors accept no liability, 0 nada for the use of this software.  It is offered "As IS"  Have fun with it!!
+ */
+
+package cloudclient
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/websocket"
+	"github.com/nats-io/nats.go"
+	log "github.com/sirupsen/logrus"
+	"github.com/theotw/natssync/pkg"
+	"github.com/theotw/natssync/pkg/bridgemodel"
+	v1 "github.com/theotw/natssync/pkg/bridgemodel/generated/v1"
+	"github.com/theotw/natssync/pkg/msgs"
+	"net/url"
+	"strings"
+)
+
+// WebSocketMessageHandler A web socket based implementation of the BiDiMessageHanadler
+type WebSocketMessageHandler struct {
+	serverURL string
+}
+
+func NewWebSocketMessageHandler(serverURL string) *WebSocketMessageHandler {
+	ret := new(WebSocketMessageHandler)
+	ret.serverURL=serverURL
+	return ret
+}
+func (t *WebSocketMessageHandler) GetHandlerType() string{
+	return "web-socket"
+}
+func (t *WebSocketMessageHandler) StartMessageHandler(clientID string) error {
+	urlSplit := strings.SplitAfterN(t.serverURL, "://", 2)
+	urlObject := url.URL{
+		Scheme: "ws",
+		Host:   urlSplit[1],
+		Path:   fmt.Sprintf("/bridge-server/1/message-queue/%s/ws", clientID),
+	}
+	websocketURL := urlObject.String()
+	log.WithField("websocketURL", websocketURL).Info("Using websocket transport")
+
+	conn, _, err := websocket.DefaultDialer.Dial(websocketURL, nil)
+	if err != nil {
+		log.WithError(err).WithField("url", websocketURL).Error("Failed to connect to websocket")
+		return err
+	}
+	go t.subscribeAndSendMessageToCloud(conn,clientID)
+	go t.ReadWSFromCloud(conn)
+	return nil
+}
+func (t *WebSocketMessageHandler) StopMessageHandler() {
+	//TODO Should probably do something here
+}
```

Comment:
Should we be unsubscribing from the subject here, like in the REST message handler?